README.md

```diff
@@ -1,135 +1,67 @@
+- `VolumeLoaderContext` is an interface that lets you initialize asynchronous data loading of different formats via its `createLoader` method
+- `IVolumeLoader` is an interface for requesting volume dimensions and data.
+- `LoadSpec` is a small bundle of information to guide the IVolumeLoader on exactly what to load.
 
-It also provides the following two utility modules:
+There are several ways to deliver volume data to the viewer:
 
-- `VolumeLoader` is a convenience class for downloading and unpacking texture atlases from .png files (up to 3 channels per png) into a Volume.
-- `VolumeMaker` is a convenience module for creating simple test volume data
-
-There are two ways to deliver volume data to the viewer:
-
-- raw Uint8Arrays of 3d volume data (one Uint8Array per channel). ( `Volume.setChannelDataFromVolume` )
-- texture atlases (png files or Uint8Arrays containing volume slices tiled across a 2d image) ( `Volume.setChannelDataFromAtlas` )
+- load OME-Zarr as publicly accessible web links. Authentication is not explicitly supported in Vol-E.
+- raw TypedArrays of 3d volume data ( see `RawArrayLoader` and `Volume.setChannelDataFromVolume` )
+- (legacy) texture atlases as png files or Uint8Arrays containing volume slices tiled across a 2d image ( see `JsonImageInfoLoader` and `Volume.setChannelDataFromAtlas` )
 
 # Example
 
-See public/index.ts for a working example. (`npm install; npm run dev` will run that code) The basic code to get the volume viewer up and running is as follows:
+See public/index.ts for a working example. (`npm install; npm run dev` will run that code)
 
-```javascript
-import { View3d, Volume, VolumeLoader, VolumeMaker } from "volume-viewer";
+The basic code to get the volume viewer up and running is as follows:
 
+```javascript
 // find a div that will hold the viewer
-const el = document.getElementById("volume-viewer");
+const el = document.getElementById("vol-e");
+
+// create the loaderContext
+const loaderContext = new VolumeLoaderContext(CACHE_MAX_SIZE, CONCURRENCY_LIMIT, PREFETCH_CONCURRENCY_LIMIT);
 
 // create the viewer.  it will try to fill the parent element.
 const view3D = new View3d(el);
+view3D.loaderContext = loaderContext;
 
-// create a volume image with dimensions passed in via jsondata
-// this json format is documented in Volume.ts as ImageInfo
-const aimg = new Volume(jsondata);
+// ensure the loader worker is ready
+await loaderContext.onOpen();
+// get the actual loader.  In most cases this will create a WorkerLoader that uses a OmeZarrLoader internally.
+const loader = await loaderContext.createLoader(path);
 
+const loadSpec = new LoadSpec();
+// give the loader a callback to call when it receives channel data asynchronously
+const volume = await loader.createVolume(loadSpec, (v: Volume, channelIndex: number) => {
+  const currentVol = v;
+
+  // currently, this must be called!
```

Comment:
This must be called when? Every time data is loaded?

Reply:
yes! ♾️